### PR TITLE
Update Mage_Eav_Model_Entity_Collection_Abstract::joinTable()

### PR DESCRIPTION
**Pull request by cebe**
### Description (*)

allow passing a `Zend_Db_Expr` object.

Zend_Db_Select::join*() methods accept `Zend_Db_Expr` for a table name, which allows joining a subquery instead of a table by name.

When passing such an object through `Mage_Eav_Model_Entity_Collection_Abstract::joinTable()` there is an edge case where it fails on this line when the subquery contains a `/`. Adding the `is_object()` check makes this more robust.

### Related Pull Requests

none.

### Fixed Issues (if relevant)

none.

### Manual testing scenarios (*)

unfortunately this is a very specific edge case so there is no manual testing.
Here is an example call on how a subquery can be used:

```php
        $subQuery = new Zend_Db_Expr("(" .
            "SELECT request_path, category_id FROM core_url_rewrite WHERE is_system = 1 ".
            " AND " . $this->_connection->quoteInto('store_id = ?', $storeId).
            " AND " . $this->_connection->quoteInto('id_path LIKE ?', 'category/%').
        ")");
        $collection->joinTable(
            array('url_rewrite' => $subQuery),
            'category_id=entity_id',
            array('request_path'),
            null,
            'left'
        );
```

If you need more information, I am happy to provide it.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Comment by elidrissidev** [comment]
I'd also update the method's docblock to better indicate it supports `Zend_Db_Expr` as a table name

**Comment by cebe** [comment]
> I'd also update the method's docblock to better indicate it supports `Zend_Db_Expr` as a table name

done.

**Review by elidrissidev — APPROVED** [review]

**Review by ADDISON74 — APPROVED** [review]

File: app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php [hunk 1]
```diff
@@ -772,7 +772,7 @@ public function joinTable($table, $bind, $fields = null, $cond = null, $joinType
         }
 
         // validate table
-        if (strpos($tableName, '/') !== false) {
+        if (!is_object($tableName) && strpos($tableName, '/') !== false) {
```
Comment (kiatng) [inline app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php:775]: I prefer a stricter check with `instanceof Zend_Db_Expr`

Comment (fballiano) [reply app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php:775]: actaully I don't understand why the original code was `if (strpos($tableName, '/') !== false)`, I wouldn't even check for anything, just that `$tableName` is a string

Comment (cebe) [reply app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php:775]: @kiatng @fballiano like this? checking for string before making a string comparision seems to be the most resonable thing to do:

```suggestion
        if (is_string($tableName) && strpos($tableName, '/') !== false) {
```

Comment (elidrissidev) [reply app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php:775]: @fballiano If the table name contains a `/` it means it's a resource name like `core/url_rewrite`, so it should be converted to the raw table name `core_url_rewrite`.

**Comment by elidrissidev** [comment]
Can you also resolve the conflicts, please.

File: app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php (hunk 1 above)
Comment (fballiano) [reply app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php:775]: right!

**Review by kiatng — DISMISSED** [review]

**Comment by fballiano** [comment]
> Can you also resolve the conflicts, please.

I've done that

**Review by fballiano — APPROVED** [review]

**Review by ADDISON74 — APPROVED** [review]

**Review by elidrissidev — APPROVED** [review]

**Pull request merged**